### DISCUSSION TITLE
Snap to Grid to use center of node instead of top left corner.

### DISCUSSION THREAD

**Discussion by JeremyGEA** [Blazor-Diagrams/Blazor.Diagrams#240] [Q&A] [answered]
I am trying to use this library to make editable HMI's for I/O control. So far, it has been going great except for the very annoying path routing using Orthogonal Link Routing. I don't think it is an issue with the orthogonal path router but rather the nodes are not snapping to the grid at the ports (left, right, top, bottom, not corners). So if my grid is set to 25px and I have a node that is 30px and another that is 45px side by side, I cannot get the links to be straight between them.

Could there be an option to snap to the center of the Node instead of top left corner?

Note: I updated to 3.0.0 beta2. Also, great work on the library! I added blazor to my WPF application in order to use this awesome diagramming library!

**Comment by zHaytam** [comment]
Hello, thank you for the kind words.

Do you have a picture showing the issue? From the explanation alone, I don't really get it sorry.
Also, do you actually need an orthogonal router (with pathfinder) or just a simple one? The current one is really not optimized and I'm thinking of droping the implementation until I find a better one, or add another implementation without the pathfinder.

**Reply by JeremyGEA** [reply]
I find the orthogonal router works well and yes, I do need it since I'm sometimes connecting nodes from different y-levels. 

Here is a picture showing 3 nodes of different heights that I'm trying to connect. 
![nodes with varying heights](https://user-images.githubusercontent.com/49650352/190209380-1e31e26e-4e84-41a7-a73d-a022665f8542.png)
Because the ports are at the middle of each node's height, the connecting lines will never line up. If we had an option to snap to the center of the node, the top, bottom, left, and right nodes would all line up with the grid.

**Comment by JeremyGEA** [comment, marked as answer]
Tada! If I change the library to use NodeModel instead of MovableModel in order to gain access to node size and use the size on pointer down and subtract the middle point when setting position, it snaps the nodes to the center instead:
![image](https://user-images.githubusercontent.com/49650352/190448563-ad260378-5500-4db4-b5dd-cc304e4fbb2a.png)

I have a PR for you: https://github.com/Blazor-Diagrams/Blazor.Diagrams/pull/241
